app/models/king.rb

```diff
@@ -1,15 +1,98 @@
 class King < Piece
+
+  def y_position
+    y_position = 0
+    if self.get_color == WHITE
+      y_position = 7
+    end
+    y_position
+  end  
   
   def valid_move?(destination_x, destination_y)
     valid = super(destination_x, destination_y)
     if valid
-      if (destination_x - self.x_position).abs > 1 || (destination_y - self.y_position).abs > 1
+      if can_castle?(destination_x, destination_y)
+        return true
+      elsif (destination_x - self.x_position).abs > 1 || (destination_y - self.y_position).abs > 1
         valid = false
       end
     end
     return valid
   end
 
+  
+
+# CASTLING KING
+
+  def move_to(destination_x, destination_y)
+    valid = super
+    if valid
+      castle!(destination_x, destination_y) 
+    end
+    valid
+  end
+
+  def can_castle?(destination_x, destination_y)
+    if self.moves == 0 && (castling_kingside?(destination_x, destination_y) || 
```

Comment:
FYI if you remove an `if` from an if statement, it becomes a boolean. This means you can return it directly:
```
if my_condition
    return true
end
# VS
return my_condition
```